### PR TITLE
Make pageSize configurable

### PR DESCRIPTION
**Pull request by fwantastic**
This will address the issue #1796. A new property called azkaban.display.numOfExecutions will be added in the azkaban.properties and be used to determine the number of rows to be displayed.

As it will default to 16 when the property does not exist, it will not have any impact on the existing page size.

### CODE REVIEW

**Comment by HappyRay** [comment]
Thanks.

Please see the config key naming conventions at 
https://github.com/HappyRay/azkaban/blob/98497ede72ee47ddabaa2d57038171f6161e2040/az-core/src/main/java/azkaban/Constants.java#L34

**Comment by jh3155** [comment]
Names have been changed to follow the naming convention!

**Comment by jh3155** [comment]
Thanks for the feedback. 
I do like what you suggested - azkaban.display.execution_page_size. I went for azkaban.page.size.xxx as if there's another page that will have non-execution rows and needs a different page size.

I'll check out the SaveAction plugin, I thought it only supported intellij and I'm using eclipse but I guess I was wrong!

**Comment by jh3155** [comment]
Installed intellij and applied the style sheet, though it didn't add a new line at the end so I manually added one. Also refactor - rename variable doesn't work properly so I had to rename them one by one as well!

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java [hunk 1]
```diff
@@ -106,6 +108,7 @@ public void init(final ServletConfig config) throws ServletException {
     this.label = props.getString("azkaban.label", "");
     this.color = props.getString("azkaban.color", "#FF0000");
     this.passwordPlaceholder = props.getString("azkaban.password.placeholder", "Password");
+    this.displayExecutionPageSize = props.getInt("azkaban.display.execution_page_size", 16);
```
Comment (HappyRay) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java]: Please define new config keys in the Constants class.

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java [hunk 2]
```diff
@@ -320,6 +323,8 @@ protected Page newPage(final HttpServletRequest req, final HttpServletResponse r
       page.add("triggerPlugins", this.triggerPlugins);
     }
 
+    page.add("displayExecutionPageSize", displayExecutionPageSize);
```
Comment (HappyRay) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java]: Make the key "displayExecutionPageSize" a constant?

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java [hunk 3]
```diff
@@ -369,4 +376,8 @@ protected void writeJSON(final HttpServletResponse resp, final Object obj, final
     resp.setContentType(JSON_MIME_TYPE);
     JSONUtils.toJSON(obj, resp.getOutputStream(), true);
   }
-}
+
+  public int getDisplayExecutionPageSize() {
+    return displayExecutionPageSize;
+  }
+}
```
Comment (HappyRay) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java]: Add new line.

**Comment by fwantastic** [comment]
keys were extracted to a constant.

**Comment by fwantastic** [comment]
hm not sure why the build is failing. any ideas?

**Comment by HappyRay** [comment]
I would download the raw travis log and look for test failures. 

How to find test failure from the ci log

Download the raw log
Open in an editor such as intellij:
Use the reg expression
Test > \w* FAILED
E.g.
Found 
azkaban.execapp.event.JobCallbackRequestMakerTest > simulateNotOKStatusCodeTest FAILED

**Comment by HappyRay** [comment]
I added the tip to https://github.com/azkaban/azkaban/wiki/Developer-Tools-and-Tips

**Comment by fwantastic** [comment]
@HappyRay the build seems to be fine, what do you think about those code coverages?

File: az-core/src/main/java/azkaban/Constants.java [hunk 4]
```diff
@@ -226,6 +226,20 @@
     public static final String QUEUEPROCESSING_ENABLED = "azkaban.queueprocessing.enabled";
 
     public static final String SESSION_TIME_TO_LIVE = "session.time.to.live";
+
+    // allowed max size of project dir in mb
+    public static final String PROJECT_DIR_MAX_SIZE = "azkaban.project_cache_max_size_in_mb";
```
Comment (HappyRay) [inline az-core/src/main/java/azkaban/Constants.java]: Why does this change show up here? This seems to belong to another change.

File: az-core/src/main/java/azkaban/Constants.java [hunk 5]
```diff
@@ -299,4 +313,9 @@
     public static final String START_TIME = "startTime";
     public static final String TRIGGER_INSTANCE_ID = "triggerInstanceId";
   }
+
+  public static class PageProperties {
+    // number of rows to be displayed on the executions page.
+    public static final String DISPLAY_EXECUTION_PAGE_SIZE = "displayExecutionPageSize";
```
Comment (HappyRay) [inline az-core/src/main/java/azkaban/Constants.java]: What's this for?

**Comment by HappyRay** [comment]
@kunkun-tang has added additional support for our new documentation system. Going forward, we would expect changes to include documentation changes as needed in the same PR whenever possible. That work is still in progress.

Could you work with him to see if it is doable now?

@kunkun-tang 
could you review this change?

File: az-core/src/main/java/azkaban/Constants.java (hunk 4 above)
Comment (fwantastic) [reply az-core/src/main/java/azkaban/Constants.java]: @HappyRay there was a merge conflict while I was updating my branch to master's head. Would it be better to close this PR and open a new one?

**Comment by fwantastic** [comment]
@HappyRay I'm happy to provide documentations for the changes. The last commit I made was for the merge conflicts that CI was flagging about. This PR's changes are pretty simple and straightforward.

**Comment by fwantastic** [comment]
to avoid any confusions, I've reverted the merge.

**Comment by fwantastic** [comment]
@HappyRay bump!

**Comment by HappyRay** [comment]
Please work with @kunkun-tang

**Comment by fwantastic** [comment]
@kunkun-tang could you review the changes please?!

**Comment by kunkun-tang** [comment]
@fwantastic Please rebase from the latest upstream master. Will start reviewing it from there. Thanks!

**Comment by fwantastic** [comment]
@kunkun-tang I've rebased the branch and should be ready for review. I accidently committed a blank file so I removed it in the last commit

File: az-core/src/main/java/azkaban/Constants.java (hunk 5 above)
Comment (kunkun-tang) [reply az-core/src/main/java/azkaban/Constants.java]: @fwantastic The same comment as described for DISPLAY_EXECUTION_PAGE_SIZE? Could you please re-elaborate this constant?

File: azkaban-solo-server/src/main/resources/conf/azkaban.properties [hunk 6]
```diff
@@ -41,3 +41,5 @@ jetty.connector.stats=true
 executor.connector.stats=true
 # Azkaban plugin settings
 azkaban.jobtype.plugin.dir=plugins/jobtypes
+#number of executions to be displayed
+azkaban.display.execution_page_size=100
```
Comment (kunkun-tang) [inline azkaban-solo-server/src/main/resources/conf/azkaban.properties]: Isn't it too large for default solo server?

File: az-core/src/main/java/azkaban/Constants.java [hunk 7]
```diff
@@ -229,6 +229,9 @@
 
     // allowed max size of shared project dir in MB
     public static final String PROJECT_DIR_MAX_SIZE_IN_MB = "azkaban.project_cache_max_size_in_mb";
+
+    // number of rows to be displayed on the executions page.
+    public static final String DISPLAY_EXECUTION_PAGE_SIZE = "azkaban.display.execution_page_size";
```
Comment (kunkun-tang) [inline az-core/src/main/java/azkaban/Constants.java]: This should be removed, right?

File: azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java [hunk 8]
```diff
@@ -369,4 +379,8 @@ protected void writeJSON(final HttpServletResponse resp, final Object obj, final
     resp.setContentType(JSON_MIME_TYPE);
     JSONUtils.toJSON(obj, resp.getOutputStream(), true);
   }
+
+  public int getDisplayExecutionPageSize() {
```
Comment (kunkun-tang) [inline azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java]: We may don't need to introduce `public` here? package-access is good enough?

File: azkaban-web-server/src/main/resources/azkaban/webapp/servlet/velocity/flowpage.vm [hunk 9]
```diff
@@ -48,6 +48,8 @@
     var projectName = "${project.name}";
     var flowId = "${flowid}";
     var execId = null;
+
+    var displayExecutionPageSize = "${displayExecutionPageSize}";
```
Comment (kunkun-tang) [inline azkaban-web-server/src/main/resources/azkaban/webapp/servlet/velocity/flowpage.vm]: Why it introduces an extra blank line?

**Comment by kunkun-tang** [comment]
Thanks for the contribution!

**Comment by kunkun-tang** [comment]
Hey @fwantastic , have you tested this feature before?

**Comment by fwantastic** [comment]
hey @kunkun-tang, I've tested this manually on my machine but there's no test for it. was there any issues with this?

**Comment by kunkun-tang** [comment]
Yes. It turned out the page still missed execution entires when it is being initialized. I'm able to reproduce the issue in my local solo server.

Per my understandg , Before {size} is fully loaded, the javascript started to call it.

**Comment by kunkun-tang** [comment]
I think I found the issue. We need to add "page" parameter in 
```
    page.add("size", getDisplayExecutionPageSize());
```
in AbstractAzkabanServlet class. @fwantastic

**Comment by fwantastic** [comment]
@kunkun-tang but that is already added in HistoryServlet's handleHistoryPage method
at line 139:
```
page.add("size", pageSize);
```

**Comment by kunkun-tang** [comment]
That is for History Page. Looks like flow page also needs. @fwantastic

**Comment by fwantastic** [comment]
@kunkun-tang Oh I was dumb.. I thought you were talking about the history page this whole time. Yes I totally forgot that this PR covers the flow page as well and I shouldn't have removed the size property in abstract servlet class. You are correct.

**Pull request merged**